modules/code-builder/docs/PRD.md

```diff
@@ -0,0 +1,128 @@
+
+# Product Requirements Document (PRD): Masonry Framework
+
+## 1. Bricks
+
+### a. Brick Types
+
+#### 1. Data Bricks: These serve as inputs for other bricks and come in two types: hardcoded and editable.
+   - Hardcoded Data Brick: Fixed values that cannot be changed by the user. Examples include predefined note values, counts, etc.
+
+     ![alt text](./images/image.png)
+   - Editable Data Brick: Values that can be modified by the user. When clicked, these bricks open a text editor or a dropdown menu for user input, allowing customization of note names, pitches, etc.
```

Comment:
Visual cues, maybe add a screenshot of a scratch/blockly brick that has a textbox, dropdown